File: src/java/inventory-ordering-service/src/main/java/com/product/pricing/core/events/internal/ProductUpdatedEvent.java
package com.product.pricing.core.events.internal;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class ProductUpdatedEvent {
    private String productId;
    private ProductDetails previous;
    private ProductDetails updated;

    public ProductUpdatedEvent(){
        this.productId = "";
        this.previous = new ProductDetails();
        this.updated = new ProductDetails(); 
    }

    public ProductUpdatedEvent(String productId, ProductDetails previousDetails, ProductDetails newDetails) {
        this.productId = productId;
        this.previous = previousDetails;
        this.updated = newDetails;
    }

    public String getProductId() {
        return productId;
    }

    public ProductDetails getPrevious() {
        return previous;
    }

    public ProductDetails getUpdated() {
        return updated;
    }
}

File: src/dotnet/src/Product.Api/ProductApi.Core/ProductUpdatedEvent.cs
namespace ProductApi.Core;

public record ProductUpdatedEvent
{
    public ProductUpdatedEvent(Product product)
    {
        this.ProductId = product.ProductId;
        this.Previous = product.PreviousDetails;
        this.Updated = product.Details;
    }
    
    public string ProductId { get; set; }

    public ProductDetails? Previous { get; set; }

    public ProductDetails Updated { get; set; }
}